<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>630</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Editor[*]</string>
  </property>
  <property name="iconSize">
   <size>
    <width>16</width>
    <height>16</height>
   </size>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="centralWidgetLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="normalPage">
       <layout class="QVBoxLayout" name="normalPageLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="tabPosition">
           <enum>QTabWidget::North</enum>
          </property>
          <property name="tabShape">
           <enum>QTabWidget::Rounded</enum>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <property name="documentMode">
           <bool>true</bool>
          </property>
          <widget class="CodeTab" name="codeTab">
           <attribute name="title">
            <string>Code</string>
           </attribute>
          </widget>
          <widget class="TilesTab" name="tilesTab">
           <attribute name="title">
            <string>Tiles</string>
           </attribute>
          </widget>
          <widget class="LevelsTab" name="levelsTab">
           <attribute name="title">
            <string>Levels</string>
           </attribute>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="debugPage"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>ToolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>16</width>
     <height>16</height>
    </size>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSaveAll"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="separator"/>
   <addaction name="actionCut"/>
   <addaction name="actionCopy"/>
   <addaction name="actionPaste"/>
   <addaction name="separator"/>
   <addaction name="actionRun"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>630</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionSaveAll"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionSelectAll"/>
    <addaction name="actionClearSelection"/>
    <addaction name="separator"/>
    <addaction name="actionGoToLine"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>&amp;Debug</string>
    </property>
    <addaction name="actionRun"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuDebug"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/door_in.png</normaloff>:/resources/fatcow16x16/door_in.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/undo.png</normaloff>:/resources/fatcow16x16/undo.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/redo.png</normaloff>:/resources/fatcow16x16/redo.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/lightbulb_off.png</normaloff>:/resources/fatcow16x16/lightbulb_off.png</iconset>
   </property>
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/tango16x16/edit-cut.png</normaloff>:/resources/tango16x16/edit-cut.png</iconset>
   </property>
   <property name="text">
    <string>Cu&amp;t</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/page_copy.png</normaloff>:/resources/fatcow16x16/page_copy.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/paste_plain.png</normaloff>:/resources/fatcow16x16/paste_plain.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/select_by_adding_to_selection.png</normaloff>:/resources/fatcow16x16/select_by_adding_to_selection.png</iconset>
   </property>
   <property name="text">
    <string>Select &amp;all</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionClearSelection">
   <property name="text">
    <string>Cl&amp;ear selection</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/tango16x16/media-playback-start.png</normaloff>:/resources/tango16x16/media-playback-start.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Run...</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/cross.png</normaloff>:/resources/fatcow16x16/cross.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionSaveAll">
   <property name="icon">
    <iconset resource="res/resources.qrc">
     <normaloff>:/resources/fatcow16x16/disk_multiple.png</normaloff>:/resources/fatcow16x16/disk_multiple.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save all</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionGoToLine">
   <property name="text">
    <string>&amp;Go to line...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeTab</class>
   <extends>QWidget</extends>
   <header>code/CodeTab.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TilesTab</class>
   <extends>QWidget</extends>
   <header>tiles/TilesTab.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LevelsTab</class>
   <extends>QWidget</extends>
   <header>levels/LevelsTab.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
